spec/middlewares/redirect_to_final_game_page_spec.rb

```diff
@@ -0,0 +1,53 @@
+RSpec.describe RedirectToFinalGamePage do
+  def app
+    Rack::Builder.parse_file('config.ru').first
+  end
+
+  context 'when won game' do
+    include_context 'when game win'
+
+    it 'game redirects from lose' do
+      get(Constants::LOSE_GAME_PATH)
+
+      expect(last_response).to be_redirect
+      expect(last_response.header['Location']).to eq(Constants::WIN_GAME_PATH)
+    end
+
+    it 'game redirects from game' do
+      get(Constants::GAME_PATH)
+
+      expect(last_response).to be_redirect
+      expect(last_response.header['Location']).to eq(Constants::WIN_GAME_PATH)
+    end
+
+    it 'new game can be started after win' do
+      get(Constants::HOME_PATH)
+
+      expect(last_response.body).to include('Start the game!')
+    end
+  end
+
+  context 'when lose-game' do
+    include_context 'when game lose'
+
+    it 'game redirects from win' do
+      get(Constants::WIN_GAME_PATH)
+
+      expect(last_response).to be_redirect
+      expect(last_response.header['Location']).to eq(Constants::LOSE_GAME_PATH)
+    end
+
+    it 'game redirects from game' do
+      get(Constants::GAME_PATH)
+
+      expect(last_response).to be_redirect
+      expect(last_response.header['Location']).to eq(Constants::LOSE_GAME_PATH)
+    end
+
+    it 'new game can be started after lose' do
+      get(Constants::HOME_PATH)
+
+      expect(last_response.body).to include('Start the game!')
```

Comment:
Instead of string it would be better to use I18n, constant or value from let variable

Reply:
+